cli/utils.js

```diff
@@ -190,17 +190,8 @@ const done = errorMsg => {
 
 const createYeomanEnv = () => {
     const env = yeoman.createEnv();
-    /* Register yeoman generators */
-    Object.keys(SUB_GENERATORS)
-        .filter(command => !SUB_GENERATORS[command].cliOnly)
-        .forEach(generator => {
-            if (SUB_GENERATORS[generator].blueprint) {
-                /* eslint-disable prettier/prettier */
-                env.register(require.resolve(`${SUB_GENERATORS[generator].blueprint}/generators/${generator}`), `${CLI_NAME}:${generator}`);
-            } else {
-                env.register(require.resolve(`../generators/${generator}`), `${CLI_NAME}:${generator}`);
-            }
-        });
+    // Lookup for all available generators (blueprints, modules, ...)
+    env.lookup();
```

Comment:
How does it know to register sub gens from blueprints? I wasn't aware of this API, is it new?

Reply:
I don't think it's new, this is what yeoman uses with the yo command.
It scans local and global node_modules for generators, and register them and their subgenerators. If a generator is available both at local and global level, local one has priority.

Reply:
I think lookup should be called only if blueprints is passed.
Otherwise it's not used at all.

Reply:
This is also necessary for modules hooks

Reply:
Calling env.lookup() first time composeExternalModules/composeBlueprint will do the job?

Reply:
No, this is required for checks and version retrieval.